src/app/api.ts

```diff
@@ -1075,6 +1108,98 @@ export function newAPI(): API {
     return secrets;
   }
 
+  async function createSecretProjects(
```

Comment:
I think you could reuse more code between Project and ProjectGroup apis since the unique difference is the path prefix.

Reply:
I haven't said to use an unique function, but to share more code between them.